delete-user-data/functions/src/helpers.ts

```diff
@@ -9,3 +13,195 @@ export const getDatabaseUrl = (
 
   return `https://${selectedDatabaseInstance}.${selectedDatabaseLocation}.firebasedatabase.app`;
 };
+
+export async function deleteCollection(db, collectionPath, batchSize) {
+  const collectionRef = db.collection(collectionPath);
+  const query = collectionRef.orderBy("__name__").limit(batchSize);
+
+  return new Promise((resolve, reject) => {
+    deleteQueryBatch(db, query, resolve).catch(reject);
+  });
+}
+
+async function deleteQueryBatch(db, query, resolve) {
+  const snapshot = await query.get();
+
+  const batchSize = snapshot.size;
+  if (batchSize === 0) {
+    // When there are no documents left, we are done
+    resolve();
+    return;
+  }
+
+  // Delete documents in a batch
+  const batch = db.batch();
+  snapshot.docs.forEach((doc) => {
+    batch.delete(doc.ref);
+  });
+  await batch.commit();
+
+  // Recurse on the next process tick, to avoid
+  // exploding the stack.
+  process.nextTick(() => {
+    deleteQueryBatch(db, query, resolve);
+  });
+}
+
+export async function repeat(
+  fn: { (): Promise<any>; (): any },
+  until: { ($: any): any; (arg0: any): any },
+  retriesLeft = 5,
+  interval = 1000
+) {
+  const result = await fn();
+
+  if (!until(result)) {
+    if (retriesLeft) {
+      await new Promise((r) => setTimeout(r, interval));
+      return repeat(fn, until, retriesLeft - 1, interval);
+    }
+    throw new Error("Max repeats count reached");
+  }
+
+  return result;
+}
+
+export const waitForDocumentToExistWithField = (
```

Comment:
is this used?

Reply:
It doesn't seem to be required. Removed.